frontends/web/src/routes/account/send/send.tsx

```diff
@@ -339,32 +340,8 @@ class Send extends Component<Props, State> {
         this.convertToFiat(result.amount.amount);
       }
     } else {
-      const errorCode = result.errorCode;
-      switch (errorCode) {
-      case 'invalidAddress':
-        this.setState({ addressError: this.props.t('send.error.invalidAddress') });
-        break;
-      case 'invalidAmount':
-      case 'insufficientFunds':
-        this.setState({
-          amountError: this.props.t(`send.error.${errorCode}`),
-          proposedFee: undefined,
-        });
-        break;
-      case 'feeTooLow':
-        this.setState({ feeError: this.props.t('send.error.feeTooLow') });
-        break;
-      case 'feesNotAvailable':
-        this.setState({ feeError: this.props.t('send.error.feesNotAvailable') });
-        break;
-      default:
-        this.setState({ proposedFee: undefined });
-        if (errorCode) {
-          this.unregisterEvents();
-          alertUser(errorCode, { callback: this.registerEvents });
-        }
-      }
-      this.setState({ isUpdatingProposal: false });
+      const errorHandling = txProposalErrorHandling(this.registerEvents, this.unregisterEvents, result.errorCode);
+      this.setState({ ...errorHandling, isUpdatingProposal: false });
```

Comment:
I was thinking what it would take to not need to pass `this.registerEvents`, `this.unregisterEvents`. I guess if `txProposalErrorHandling` would return a promise, it could resolve or we could reject on default+errorCode and then move the alertUser out of the promise...

I'm not sure, maybe for a future refactor?

Reply:
Yes let's do it as a future refactor!